### PR TITLE
Change bn to u64 to match back-end

### PR DESCRIPTION
**Pull request by rhubarbgarden**
Also added calculation for amount of votes per share in governance

### CODE REVIEW

**Comment by callensm** [comment]
Is this meant to just be a semantic change? `BN` and `u64` are functionally the same.

File: src/staking/stakeAccount.ts [hunk 1]
```diff
@@ -118,15 +119,24 @@ export class StakeAccount {
...
    */
   static useBalance(stakeAccount?: StakeAccount, stakePool?: StakePool): StakeBalance {
-    const jetPerStakedShare = bnToNumber(stakePool?.vault.amount.div(stakePool?.stakePool.sharesBonded))
+    let stakedJet: u64 = new u64(0)
+    let unbondingJet: u64 = new u64(0)
+    let jetVotesPerShare: u64 = new u64(0)
 
-    const stakedJet = bnToNumber(stakeAccount?.stakeAccount.shares) * jetPerStakedShare
+    if (!!stakePool && !!stakeAccount) {
+      jetVotesPerShare = stakePool.vault.amount.div(stakePool.stakePool.sharesBonded)
```
Comment (Tristyn) [inline src/staking/stakeAccount.ts:127]: `stakedJet` looks right, but I think this one would have a division rounding issue, do you need to multiply it by another BN first then divide last to get the votes balance?

**Review by Tristyn — CHANGES_REQUESTED** [review]
The useBalance change is cool

I would not use u64 over BN because of a couple issues

**Comment by Tristyn** [comment]
> Is this meant to just be a semantic change? `BN` and `u64` are functionally the same.
> 
> Only adds two `Buffer` related functions to the class definitions:
> 
> ```ts
>  export class u64 extends BN {
>     toBuffer(): Buffer;
>     static fromBuffer(buffer: Buffer): u64;
>   }
> ```

It ensures buffers are 8 bytes, but anchor is smart enough to handle any sized buffers. u64 came from spl-token before anchor was a thing

File: src/staking/stakeAccount.ts (hunk 1 above)
Comment (rhubarbgarden) [reply src/staking/stakeAccount.ts:127]: `jetVotesPerShare` will be a different calculation than `stakedJet` - the program will panic if trying to mint less than one share's worth of votes (even if that is what the user is owed) so jetVotesPerShare calculates what that number of votes will be

File: src/staking/stakeAccount.ts [hunk 2]
```diff
@@ -118,15 +119,24 @@ export class StakeAccount {
    * @memberof StakeAccount
    */
   static useBalance(stakeAccount?: StakeAccount, stakePool?: StakePool): StakeBalance {
-    const jetPerStakedShare = bnToNumber(stakePool?.vault.amount.div(stakePool?.stakePool.sharesBonded))
+    let stakedJet: u64 = new u64(0)
```
[the remark this answers is not in the document]
Comment (rhubarbgarden) [reply src/staking/stakeAccount.ts:122]: Will change this to allow for undefined values before accounts have loaded

**Review by Tristyn — APPROVED** [review]
Looks good to merge, you might run into trouble using the jetPerStakedShare because it gets rounded to a whole number

File: src/staking/stakeAccount.ts (hunk 1 above)
Comment (Tristyn) [reply src/staking/stakeAccount.ts]: Maybe we should skip minting votes if there's not enough room to mint

Comment (rhubarbgarden) [reply src/staking/stakeAccount.ts]: The way I have it implemented rn is that if the value of the remaining votes is less than one share, we don't mint anything at all. Is that similar to what you were thinking?

File: src/staking/stakePool.ts [hunk 3]
```diff
@@ -98,8 +101,9 @@ export class StakePool {
     const collateralMint = parseMintAccount(collateralMintInfo.data as Buffer)
     const vault = parseTokenAccount(vaultInfo.data as Buffer, addresses.stakePoolVault)
     const tokenMint = parseMintAccount(tokenMintInfo?.data as Buffer)
+    const jetVotesPerShare = vault.amount.div(stakePool.sharesBonded)
```
[the remark this answers is not in the document]
Comment (rhubarbgarden) [reply src/staking/stakePool.ts:104]: I'm not entirely sure I understand, could you elaborate?

**Pull request merged**